tests/rptest/tests/node_pool_migration_test.py

```diff
@@ -0,0 +1,278 @@
+
+        # Start a new consumer to read all data written
+        self.start_consumer()
+        self.consumer.wait()
+
+        assert self.consumer.consumer_status.validator.invalid_reads == 0, f"Invalid reads in topic: {self._topic}, invalid reads count: {self.consumer.consumer_status.validator.invalid_reads}"
+
+    def _replicas_per_node(self):
+        kafkacat = KafkaCat(self.redpanda)
+        node_replicas = {}
+        md = kafkacat.metadata()
+        self.redpanda.logger.debug(f"metadata: {md}")
+        for topic in md['topics']:
+            for p in topic['partitions']:
+                for r in p['replicas']:
+                    id = r['id']
+                    if id not in node_replicas:
+                        node_replicas[id] = 0
+                    node_replicas[id] += 1
+
+        return node_replicas
+
+    @cluster(num_nodes=11, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    @matrix(balancing_mode=['off', 'node_add'])
+    def test_migrating_redpanda_nodes_to_new_pool(self, balancing_mode):
+        initial_pool = self.redpanda.nodes[0:5]
+        new_pool = self.redpanda.nodes[5:]
+        self.redpanda.set_seed_servers(initial_pool)
+
+        # start redpanda on initial pool of nodes
+        self.redpanda.start(nodes=initial_pool,
+                            auto_assign_node_id=True,
+                            omit_seeds_on_idx_one=False)
+        self.admin.patch_cluster_config(
+            upsert={"partition_autobalancing_mode": balancing_mode})
+        self._create_topics()
+
+        self.start_producer()
+        self.start_consumer()
+        # add new nodes to the cluster
+        for n in new_pool:
+            self.redpanda.start_node(n)
+
+        def all_nodes_present():
+            for n in self.redpanda.nodes:
+                brokers = self.admin.get_brokers(node=n)
+                return len(brokers) == len(initial_pool) + len(new_pool)
+
+        wait_until(
+            all_nodes_present,
+            60,
+            1,
+            err_msg=
+            "Not all nodes that were supposed to join the cluster are members")
+
+        for to_decommission in initial_pool:
+            to_decommission_id = self.redpanda.node_id(to_decommission)
+            self.logger.info(f"decommissioning node: {to_decommission_id}", )
+            self._decommission(to_decommission_id)
+
+        decommissioned_ids = [
+            self.redpanda.node_id(to_decommission)
+            for to_decommission in initial_pool
+        ]
+
+        self._wait_for_nodes_removed(decommissioned_ids)
+
+        self.verify()
+        r_per_node = self._replicas_per_node()
+        self.logger.info(f"finished with {r_per_node} replicas per node")
+        total_replicas = sum([r for r in r_per_node.values()])
+        tolerance = total_replicas * 0.1
+        min_expected = total_replicas / len(new_pool) - tolerance
+        max_expected = total_replicas / len(new_pool) + tolerance
+
+        assert all(
```

Comment:
I'm wondering if this needs to be wrapped in a waiter since uniformity eventually happens across ticks.